m4/threads.m4

```diff
@@ -32,39 +32,47 @@ AC_DEFUN([ACX_BEST_THREAD],
   dnl implementations.
   dnl
   dnl [1]: https://gcc.gnu.org/onlinedocs/libgomp/OMP_005fNUM_005fTHREADS.html
-  AX_OPENMP([],[enableopenmp=no])
+  AC_ARG_ENABLE(openmp,
+                AS_HELP_STRING([--disable-openmp],
+                               [build without OpenMP support]),
+                [AS_CASE("${enableval}",
+                         [yes], [enableopenmp=yes],
+                         [no],  [enableopenmp=no],
+                         [AC_MSG_ERROR(bad value ${enableval} for --enable-openmp)])],
+                [enableopenmp=yes])
 
-  dnl The call to AX_OPENMP only sets OPENMP_CXXFLAGS, so if it worked,
-  dnl we also set it for C, Fortran, and all the METHODs.
-  AS_IF([test "x$OPENMP_CXXFLAGS" != "x"],
-        [
-          AC_MSG_RESULT(<<< Configuring library with OpenMP support >>>)
-          OPENMP_CFLAGS=$OPENMP_CXXFLAGS
-          OPENMP_FFLAGS=$OPENMP_CXXFLAGS
-          CXXFLAGS_OPT="$CXXFLAGS_OPT $OPENMP_CXXFLAGS"
-          CXXFLAGS_DBG="$CXXFLAGS_DBG $OPENMP_CXXFLAGS"
-          CXXFLAGS_DEVEL="$CXXFLAGS_DEVEL $OPENMP_CXXFLAGS"
-          CXXFLAGS_PROF="$CXXFLAGS_PROF $OPENMP_CXXFLAGS"
-          CXXFLAGS_OPROF="$CXXFLAGS_OPROF $OPENMP_CXXFLAGS"
-          CFLAGS_OPT="$CFLAGS_OPT $OPENMP_CFLAGS"
-          CFLAGS_DBG="$CFLAGS_DBG $OPENMP_CFLAGS"
-          CFLAGS_DEVEL="$CFLAGS_DEVEL $OPENMP_CFLAGS"
-          CFLAGS_PROF="$CFLAGS_PROF $OPENMP_CFLAGS"
-          CFLAGS_OPROF="$CFLAGS_OPROF $OPENMP_CFLAGS"
-          FFLAGS="$FFLAGS $OPENMP_FFLAGS"
-          AC_SUBST(OPENMP_CXXFLAGS)
-          AC_SUBST(OPENMP_CFLAGS)
-          AC_SUBST(OPENMP_FFLAGS)
-        ])
-
-  dnl If the user explicitly requested the openmp threading model and
-  dnl this compiler doesn't support OpenMP for some reason, we throw an
-  dnl error instead of silently continuing.
-  AS_IF([test "x$requested_thread_model" = "xopenmp"],
-        [
-          AS_IF([test "x$enableopenmp" = "xno"],
-                [AC_MSG_ERROR([requested openmp threading model, but compiler does not support openmp.])])
-        ])
+  AS_IF([test "x$enableopenmp" = "xyes"],
+        [AX_OPENMP([],[enableopenmp=no])]
```

Comment:
For instance, here there is a closing `]` but no comma, so I guess it doesn't start the third (else) clause of the `AS_IF` statement?

Reply:
Correct. If `enableopenmp = yes`, then we execute the `AX_OPENMP` macro and then we also execute the following macro (the `AS_IF` on whether the macro successfully set `OPENMP_CXXFLAGS`)

Reply:
The corresponding `else` for this `AS_IF` (`test "x$enableopenmp" = "xyes"`) is the `Error if the user simultaneously requested --with-thread-model=openmp and --disable-openmp` bit